### PR TITLE
chore: nitpicky clean up changes for the t9n PR

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #4961 

## Summary
A couple small nitpicks I came across when reviewing the PR. Doc consistency stuff, and changing the `commonTest` to make `index` a string once instead of twice.

Otherwise nothing else jumped out at me. It's looking good!

### CODE REVIEW

File: src/tests/commonTests.ts [hunk 1]
```diff
@@ -1022,9 +1022,9 @@ export async function t9n(
       const props: Partial<MessageBundle> = {};
 
       for (const prop of nonIntlProps) {
-        const index = nonIntlProps.indexOf(prop);
-        props[prop] = `${index}`;
-        component.setProperty(prop, `${index}`);
+        const index = nonIntlProps.indexOf(prop).toString();
```
Comment (jcfranco) [inline src/tests/commonTests.ts:1025]: Nice! ✨🧹✨

Alternatively: `${nonIntlProps.indexOf(prop)}`

This is going into a feature branch, so we don't have to split this up into a separate refactor PR. 🎉

File: conventions/Internationalization.md [hunk 2]
```diff
@@ -34,21 +34,22 @@ This pattern enables components to support built-in translations. In order to su
...
-   2. The `onMessagesChange` method must also be configured with watchers for the messages properties as well as Intl props.
+   2. The `onMessagesChange` method must also be configured with watchers for the messages properties as well as `intl` props.
 3. Use the `setUpMessages` util in the component's `componentWillLoad` lifecycle methods. This must be awaited on to have an initial set of strings available before rendering.
 4. Use the `connectMessages`/`disconnectMessages` utils in the component's `connectedCallback`/`disconnectedCallback` lifecycle methods. This will set up and tear down supporting methods on the component.
 5. Add an appropriate E2E test by using the `t9n` common test helper.
-6. Compound components ( Ex: input-date-picker) which are neither localized nor t9n need to forward `intl` props into supporting components.
-   1. mark `intl` props as optional and add `messageOverrides` prop.
-   2. Parse the `messageOverrides` along with `intl` props into supporting components in render method.
+6. Compound components ( Ex: input-date-picker) that are neither localized nor `t9n` need to forward `intl` props into supporting components.
```
Comment (jcfranco) [inline conventions/Internationalization.md:41]: Suggestion: `( Ex:` -> `(e.g.,`

**Review by jcfranco — APPROVED** [review]
Thanks!

File: src/tests/commonTests.ts (hunk 1 above)
Comment (anveshmekala) [reply src/tests/commonTests.ts:1025]: Can you update the name of nonIntlProps as `extraMessages` and do the same for the method name.

It was supposed to be updated [here ](https://github.com/Esri/calcite-components/pull/5465#discussion_r998674621) and I guess while changing the base branch the changes are wiped off.

Comment (benelan) [reply src/tests/commonTests.ts:1025]: > Can you update the name of nonIntlProps as additionalMessages and do the same for the method name.

will do!

> This is going into a feature branch, so we don't have to split this up into a separate refactor PR. 

lol I don't like randomly pushing to other people's branches. Franco be like:

personal-space

**Pull request merged**